hevea: new version 2.36 accepts ocaml 5 series

cessenat opened the pull request:
<!--  
Remember that `spackbot` can help with your PR in multiple ways:
- `@spackbot help` shows all the commands that are currently available
- `@spackbot fix style` tries to push a commit to fix style issues in this PR
- `@spackbot re-run pipeline` runs the pipelines again, if you have write access to the repository 
-->
Hevea prior to 2.35 (included) demanded ocaml 4 series.

cessenat committed:
hevea: new version 2.36 accepts ocaml 5 series

alalazo commented on var/spack/repos/builtin/packages/hevea/package.py:
```diff
@@ -28,6 +29,7 @@ class Hevea(MakefilePackage):
     # Dependency demands ocamlbuild
     depends_on("ocaml")
     depends_on("ocamlbuild")
+    conflicts("ocaml@5.0.0:", when="@:2.35")
```
Can we express that as an additional `depends_on`:
```python
depends_on("ocaml@:xxx", when="@:2.35")
```
?

cessenat replied:
Sure. Done.

cessenat committed:
Use an additional depends_on directive instead of conflicts

Replace
    conflicts("ocaml@5.0.0:", when="@:2.35")
with
    depends_on("ocaml@4.0.0:4.99.9", when="@:2.35")

alalazo commented on var/spack/repos/builtin/packages/hevea/package.py:
```diff
@@ -28,6 +29,7 @@ class Hevea(MakefilePackage):
     # Dependency demands ocamlbuild
     depends_on("ocaml")
     depends_on("ocamlbuild")
+    depends_on("ocaml@4.0.0:4.99.9", when="@:2.35")
```
This is different from the previous conflict, since it also imposes a lower bound (`ocaml` must be at major version 4). If this is the correct directive we can use just:
```suggestion
    depends_on("ocaml@4", when="@:2.35")
```
Also, shouldn't we impose a lower bound regardless of versions? It seems odd that `hevea@2.36` can use `ocaml@3` according to the directives.

cessenat replied:
That's right. BTW prior to ocaml 4.8.0 configure should not set CC={self.compiler.cc}.
Maybe I shall open a PR for that later on.

cessenat committed:
Simplify ocaml dependency

Co-authored-by: Massimiliano Culpo <massimiliano.culpo@gmail.com>

cessenat committed:
After hevea 2.34 ocaml needs to be 4.08.0 at least
